OSSM-8264 Add sections on operator & component versioning in OSSM 2 and 3 differences (#93570)
openshift/openshift-docs: modules/ossm-migrating-read-me-new-resources.adoc

> {SMProductName} 3 uses two new resources:

Reviewer: ```suggestion
{SMProductName} 3 uses the following two new resources:
```

* `{istio}` resource
* `IstioCNI` resource

[id="ossm-istio-resource-replaces-smcp_{context}"]
== The Istio resource replaces the ServiceMeshControlPlane resource

{SMProduct} 2 uses a resource called `ServiceMeshControlPlane` to configure {istio}. In {SMProduct} 3, the `ServiceMeshControlPlane` resource is replaced with a resource called `{istio}`.

The `{istio}` resource contains a `spec.values` field that derives its schema from {istio}'s Helm chart values. This means that configuration examples from the community {istio} documentation can often be applied directly to the {SMProduct} 3 `{istio}` resource.

> The `{istio}` resource provides an additional validation schema enabling the ability to explore the resource running the following the {oc-first} command:

Reviewer: ```suggestion
You can view an additional validation schema of the `{istio}` resource by running the following command:
```

IMO `{oc-first} ` is not needed here but up to you

== About Istio control plane versioning

In {SMProduct} 2.6 and earlier, the `version` field in the `ServiceMeshControlPlane` resource specified the control plane version. The `version` field accepted only minor versions, such as `v2.5` or `v2.6`. The Operator automatically applied new patch versions, such as 2.6.1, without requiring changes to the resource.

{SMProduct} 3.0 introduces the `{istio}` resource to manage {istio} control planes. This resource also includes a `version` field, but it uses {istio} versioning instead of {SMProduct} versions. The field accepts specific patch versions, such as `v1.24.1`, which the Operator maintains without applying automatic updates.

To enable automatic patch updates, use a version in the format `v1.24-latest`. This instructs the Operator to keep the {istio} control plane updated with the latest available patch release of {istio} 1.24.

[id="ossm-new-resource-istiocni_{context}"]
== New resource: IstioCNI

The {istio} Container Network Interface (CNI) node agent is used to configure traffic redirection for pods in the mesh. It runs as a daemon set, on every node, with elevated privileges.

> In {SMProduct} 2, the Operator deployed an {istio} CNI instance for each minor version of {istio} present in the cluster, and pods were automatically annotated during sidecar injection so they picked up the correct {istio} CNI. While this meant that the management of {istio} CNI was mostly hidden from you, it obscured the fact that the {istio} CNI agent has an independent lifecycle from the {istio} control plane and, in some cases, the {istio} CNI agent must be be upgraded separately.

Reviewer: ```suggestion
In {SMProduct} 2, the Operator deployed an {istio} CNI instance for each minor version of {istio} present in the cluster, and pods were automatically annotated during sidecar injection so they picked up the correct {istio} CNI. The {istio} CNI agent has an independent lifecycle from the {istio} control plane and, in some cases, you must upgrade the {istio} CNI agent separately.
```
